docs/modules/ROOT/pages/utilities.adoc

```diff
@@ -1,320 +1,289 @@
-
-[,python]
-----
-from utils import State, Account
-
-starknet = await State.init()
-account1 = await Account.deploy(public_key)
-account2 = await Account.deploy(public_key)
-----
-
-The Account class also provides access to the account contract class which is useful for following the <<memoization,Memoization>> pattern. To fetch the account contract class:
-
-[,python]
-----
-fetch_class = Account.get_class
-----
-
-== MockSigner
-
-`MockSigner` is used to perform transactions with an instance of https://github.com/OpenZeppelin/nile/blob/main/src/nile/signer.py[Nile's Signer] on a given Account, crafting the transaction and managing nonces.
-The `Signer` instance manages signatures and is leveraged by `MockSigner` to operate with the Account contract's `\\__execute__` method.
-See xref:accounts.adoc#mocksigner_utility[MockSigner utility] for more information.
+[.contract-item]
+[[utils-UnwrapAndCast]]
+==== `[.contract-item-name]#++UnwrapAndCast++#` [.item-kind]#trait#
+
+Trait for exposing an `unwrap_and_cast` function to `SyscallResult` objects. This may be useful
+when unwrapping a syscall result to a type implementing the `Serde` trait, and you want to avoid the boilerplate of
+casting and unwrapping the result multiple times.
+
+Usage example:
+
+```javascript
+use openzeppelin::utils::selectors;
+use openzeppelin::utils::UnwrapAndCast;
+
+fn call_and_cast_to_bool(target: ContractAddress, args: Span<felt252>) -> bool {
+    try_selector_with_fallback(
+        target, selectors::has_role, selectors::hasRole, args
+    ).unwrap_and_cast()
+}
+
+fn call_and_cast_to_felt252(target: ContractAddress, args: Span<felt252>) -> felt252 {
+    try_selector_with_fallback(
+        target, selectors::get_role_admin, selectors::getRoleAdmin, args
+    ).unwrap_and_cast()
+}
+```
+
+Note that it can be automatically casted to any type implementing the `Serde` trait.
+
+[#utils-Inner-Modules]
+==== Inner modules
+
+[.contract-item]
+[[utils-selectors]]
+==== `[.contract-item-name]#++selectors++#` [.item-kind]#module#
+
+See xref:#selectors[`openzeppelin::utils::selectors`].
+
+[.contract-item]
+[[utils-serde]]
+==== `[.contract-item-name]#++serde++#` [.item-kind]#module#
+
+See xref:#serde[`openzeppelin::utils::serde`].
+
+[.contract]
+[[selectors]]
+=== `++selectors++`
+
+```javascript
+use openzeppelin::utils::selectors;
+```
+
+:selectors: https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.8.0-beta.0/src/utils/selectors.cairo[selectors.cairo]
+
+Module containing constants matching multiple selectors used through the library.
+To see the full list of selectors, see {selectors}.
+
+[.contract]
+[[serde]]
+=== `++serde++`
+
+```javascript
+use openzeppelin::utils::serde;
+```
+
+Module containing utilities related to serialization and deserialization of Cairo data structures.
```

Comment:
I'm wondering if we should make `SerializedAppend` available directly in `utils.cairo` which would shorten the path to: `use openzeppelin::utils::SerializedAppend` like we're doing with `UnwrapAndCast`. Do you think we'll be adding other traits or implementations to the `serde` mod that would conflict with this approach?

Reply:
I'm tempted to agree on making it available directly, but I feel it could create an inconsistency if we add more utilities under serde at some point, and I think this could be the case in the future (or not 🥲, but I vote we keep it separated just in case). UnwrapAndCast is required for `try_selector_with_fallback` and we are certain that there are not more utilities to add there.

Reply:
Right, we _could_ add more utils to serde. Sounds good to me